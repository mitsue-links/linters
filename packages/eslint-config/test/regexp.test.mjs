import path from 'path';
import {lintFile} from './helper.mjs';

test('regexp_ok.mjs', async () => {
    const filePath = 'regexp_ok.mjs';
    const data = await lintFile(path.join('fixtures', filePath));

    expect(data).toStrictEqual([{
        filePath,
        messages: [],
    }]);
});

test('regexp_error.mjs', async () => {
    const filePath = 'regexp_error.mjs';
    const data = await lintFile(path.join('fixtures', filePath));

    expect(data).toMatchSnapshot();
});
