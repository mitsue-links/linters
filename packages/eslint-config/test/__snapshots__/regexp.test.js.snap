// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`regexp_error.js 1`] = `
Array [
  Object {
    "filePath": "regexp_error.js",
    "messages": Array [
      Object {
        "line": 5,
        "ruleId": "require-unicode-regexp",
        "severity": 2,
      },
      Object {
        "line": 13,
        "ruleId": "prefer-named-capture-group",
        "severity": 2,
      },
      Object {
        "line": 13,
        "ruleId": "prefer-named-capture-group",
        "severity": 2,
      },
      Object {
        "line": 13,
        "ruleId": "prefer-named-capture-group",
        "severity": 2,
      },
    ],
  },
]
`;
